src/Symfony/Bridge/Twig/Extension/RoutingExtension.php

```diff
@@ -69,6 +73,16 @@ public function getUrl($name, $parameters = array(), $schemeRelative = false)
         return $this->generator->generate($name, $parameters, $schemeRelative ? UrlGeneratorInterface::NETWORK_PATH : UrlGeneratorInterface::ABSOLUTE_URL);
     }
 
+    /**
+     * Check if a route exists.
+     *
+     * @see RouteCollection
+     */
+    public function hasRoute(string $name)
+    {
+        return $this->router->getRouteCollection()->has($name);
```

Comment:
calling getRouteCollection is a no-go as it is very slow
instead, I'd suggest trying to generate a route with that $name and catch the exception to return false